src/Core/Language/Pack/Loader/RemoteLanguagePackLoader.php

```diff
@@ -34,35 +36,63 @@
 final class RemoteLanguagePackLoader implements LanguagePackLoaderInterface
 {
     /**
-     * The link from which available languages are retrieved.
+     * The languages repository base path
+     *
+     * @TODO : duplicated from {prestashop_repository}/app/config/config.yml;
+     *         used for install context or other cases where no Symfony container is available
      */
-    public const PACK_LINK = 'http://i18n.prestashop-project.org/translations/%ps_version%/available_languages.json';
+    private const LANG_REPOSITORY_BASE_PATH = 'https://i18n.prestashop-project.org';
 
     /**
-     * @var Version
+     * @var Version Prestashop version
      */
     private $version;
 
     /**
-     * @param Version $version
+     * @var string represents the language repository base path
      */
-    public function __construct(Version $version)
+    private $langRepositoryBasePath;
+
+    /**
+     * @param Version $version Prestashop version
+     * @param string $langRepositoryBasePath should become mandatory in future release
+     */
+    public function __construct(Version $version, string $langRepositoryBasePath = self::LANG_REPOSITORY_BASE_PATH)
     {
         $this->version = $version;
+        $this->langRepositoryBasePath = $langRepositoryBasePath;
+    }
+
+    /**
+     * @param string $locale
+     *
+     * @return string requested URL
+     */
+    public function getLanguagePackUrl(string $locale = null): string
+    {
+        $stringVersion = $this->version->getSemVersion();
+
+        return "{$this->langRepositoryBasePath}/translations/{$stringVersion}/{$locale}/{$locale}.zip";
+    }
+
+    /**
+     * @return string requested URL
+     */
+    public function getLanguagePackListUrl(): string
+    {
+        $stringVersion = $this->version->getSemVersion();
+
+        return "{$this->langRepositoryBasePath}/translations/{$stringVersion}/available_languages.json";
     }
 
     /**
      * {@inheritdoc}
      */
     public function getLanguagePackList()
     {
-        $normalizedLink = str_replace('%ps_version%', $this->version->getSemVersion(), self::PACK_LINK);
+        $normalizedLink = $this->getLanguagePackListUrl();
         $jsonResponse = file_get_contents($normalizedLink);
-
-        $result = [];
-        if ($jsonResponse) {
-            $result = json_decode($jsonResponse, true);
-        }
+        $result = $jsonResponse ? (json_decode($jsonResponse, true) ?? []) : [];
```

Comment:
It is best to avoid nested ternaries. I think the previous version of this line was a little more readable.